errors.go

```diff
@@ -0,0 +1,50 @@
+package goanda
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+func newAPIError(request *http.Request, responce *http.Response) APIError {
+	defer responce.Body.Close()
+
+	msg := struct {
+		ErrorMessage string
+		RejectReason string
+	}{}
+
+	apiErr := APIError{
+		Responce: responce,
+		Request:  request,
+	}
+
+	b, _ := ioutil.ReadAll(responce.Body)
+	err := json.Unmarshal(b, &msg)
+	if err != nil {
+		apiErr.Message = string(b)
+	} else {
+		apiErr.Message = msg.RejectReason + msg.ErrorMessage
+	}
+	return apiErr
+}
+
+// APIError is returned when the Oanda server responds with an error
+//
+// Message is the returned error message from the server if possible to unmarshal,
+// otherwise it is simply the entire body of the responce
+type APIError struct {
+	Request  *http.Request
+	Responce *http.Response
```

Comment:
Please fix the typos, there are a few (about 15) of the "responce -> response" ones in this PR.
```suggestion
	Response *http.Response
```

Reply:
Thanks for the input! You're absolutely right, this PR is quite large. I didn't want to dive too deep into refactoring it for fear of breaking something without realising.

I've fixed the typos. Thanks for pointing them out.